bindings/wasm/src/did/wasm_method_scope.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity::did::MethodScope;
+use wasm_bindgen::prelude::*;
+
+use crate::error::Result;
+use crate::error::WasmResult;
+
+/// Supported verification method types.
+#[wasm_bindgen(js_name = MethodScope)]
+#[derive(Clone, Debug)]
+pub struct WasmMethodScope(pub(crate) MethodScope);
+
+#[wasm_bindgen(js_class = MethodScope)]
+impl WasmMethodScope {
+  #[wasm_bindgen(js_name = VerificationMethod)]
+  pub fn verification_method() -> WasmMethodScope {
+    Self(MethodScope::VerificationMethod)
+  }
```

Comment:
Could these be turned into getters, so it feels more like an enum in JS, e.g. `MethodScope.VerificationMethod` instead of `MethodScope.VerificationMethod()`? Seems preferable to me, but @eike-hass or @abdulmth might want to decide.

Reply:
Nope, the code doesn't compile with `getter` unfortunately. The alternative would be to re-create a C-style enum for the Wasm bindings and use `From`/`Into` to convert it. Then these methods would not be needed.

Reply:
Update: tried a c-style enum for `WasmMethodScope` but that breaks the `IVerifierProperties` interface, specifically the serde deserialization since in Javascript the enum maps to a `number` rather than a `WasmMethodScope`.

```rust
/// Verification method section or relationship.
#[wasm_bindgen(js_name = MethodScope)]
#[derive(Clone, Copy, Debug)]
pub enum WasmMethodScope {
  VerificationMethod,
  Authentication,
  AssertionMethod,
  KeyAgreement,
  CapabilityDelegation,
  CapabilityInvocation,
}
```

Compiles to the following (note how `MethodScope.VerificationMethod` maps to `0`):

```js
export const MethodScope = Object.freeze({
    VerificationMethod: 0,
    "0": "VerificationMethod",
    Authentication: 1,
    "1": "Authentication",
    AssertionMethod: 2,
    "2": "AssertionMethod",
    KeyAgreement: 3,
    "3": "KeyAgreement",
    CapabilityDelegation: 4,
    "4": "CapabilityDelegation",
    CapabilityInvocation: 5,
    "5": "CapabilityInvocation",
});
```

Reply:
Thanks for looking into it! I'm also fine with the method-style then. I thought there might be an easy win here, but doesn't look like it, if it breaks serialization.